### ISSUE TITLE
Adding labels/annotations to resources or data to config maps/secrets when those were empty fails

### ISSUE THREAD

**Issue by pdecat** [terraform-providers/terraform-provider-kubernetes#115]
Adding labels to a resource that didn't have any fails with a `jsonpatch add operation does not apply: doc is missing path` error message.

### Terraform Version

```
# Terraform v0.11.3
+ provider.kubernetes v1.0.1
```

Note: the exact same issue happens from master.

### Affected Resource(s)

- All resources with metadata

### Terraform Configuration Files

1. Resource without labels:

```hcl
resource "kubernetes_namespace" "test" {
	metadata {
		annotations {
			TestAnnotationOne = "one"
			TestAnnotationTwo = "two"
		}
		name = "test"
	}
}
```

2. Resource with labels:

```hcl
resource "kubernetes_namespace" "test" {
	metadata {
		annotations {
			TestAnnotationOne = "one"
			TestAnnotationTwo = "two"
		}
		labels {
			TestLabelOne = "one"
			TestLabelTwo = "two"
			TestLabelThree = "three"
		}
		name = "test"
	}
}
```

### Expected Behavior

Applying the first file then the second one should add the labels to the resource.

### Actual Behavior

Applying the first file then the second one fails:

```
                * kubernetes_namespace.test: 1 error(s) occurred:

                * kubernetes_namespace.test: jsonpatch add operation does not apply: doc is missing path: /metadata/labels/TestLabelTwo
```

### Steps to Reproduce

1. `terraform apply` with first file
2. `terraform apply` with second file

**Comment by pdecat** [comment]
Also affects annotations which use the same `diffStringMap()` function.

**Comment by pdecat** [comment]
Also affects empty config maps with no data which use the same `diffStringMap()` function.

**Comment by pdecat** [comment]
And finally, it also affects secrets with no data which use the same `diffStringMap()` function.

**Comment by radeksimko** [comment]
Fixed via #116 

Thanks for reporting this bug and sending the patch @pdecat ! 👍
